geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java

```diff
@@ -2417,9 +2417,9 @@ public int hashCode() {
     final int prime = 31;
     int result = 1;
     result = prime * result + (checkPortability ? 1231 : 1237);
-    result = prime * result + ((classPatterns == null) ? 0 : classPatterns.hashCode());
-    result = prime * result + ((excludePatterns == null) ? 0 : excludePatterns.hashCode());
-    result = prime * result + ((identityPatterns == null) ? 0 : identityPatterns.hashCode());
+    result = prime * result + classPatterns.hashCode();
```

Comment:
Is this supposed to be "classPatterns.uisEmpty() ? 0 : classPatterns.hashCode()" 
Anyways the hashCode for empty set returns 0. Not sure if there if there are any performance benefits.

Reply:
I went with thinking that hashCode implementation of HashSet is implemented by JDK and hence it must be designed in the already optimized state.

Reply:
Its also autogenerated code via the IDE